query_test.go

```diff
@@ -0,0 +1,118 @@
+package dht
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	tu "github.com/libp2p/go-libp2p-testing/etc"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TODO Debug test failures due to timing issue on windows
+// Tests  are timing dependent as can be seen in the 2 seconds timed context that we use in "tu.WaitFor".
+// While the tests work fine on OSX and complete in under a second,
+// they repeatedly fail to complete in the stipulated time on Windows.
+// However, increasing the timeout makes them pass on Windows.
+
+func TestRTEvictionOnFailedQuery(t *testing.T) {
```

Comment:
Were these tests moved from somewhere else, if so why move them?

Reply:
These are new tests. We are now testing the change in the RT state based on query responses as per the new logic we've added in _this_ PR.

Reply:
ah ok, I noticed the OS X vs Windows thing and figured it was just copy-paste from our other issue like that.

Reply:
We got rid of the old tests as they are not needed anymore/don't prove anything in light of the new changes.

However, I am sure if these tests will work on OSX/Windows (because the methodology of the tests/some of the code paths we hit are similar) and so I kept the comment as is.